mdaviz/mainwindow.py

```diff
@@ -118,51 +113,47 @@ def setmdaFileList(self,folder_path):
         self._mdaFileList = sorted([file.name for file in folder_path.glob('*.mda')])
         # TODO: what if new file gets added to the directory, you want to append those to the list without the user having to reselect the file nor the entire MVC
         # TODO: check for new file automatically (every x seconds)
-
-    # def setmdaFileList(self,folder_path, as_string=True):
-    #     if isinstance(folder_path,str):   
-    #         folder_path=Path(folder_path)
-    #     if as_string:
-    #        return sorted([file.name for file in folder_path.glob('*.mda')])
-    #     else:
-    #        return [file for file in folder_path.glob('*.mda')]
     
     def setFiles(self, files_list):
-        """Set the names (of server's catalogs) in the pop-up list."""
-        self.catalogs.clear()
-        self.catalogs.addItems(files_list)   
+        """Set the file names in the pop-up list."""
+        self.files.clear()
+        self.files.addItems(files_list)   
         
     def setFile(self,mda_file):
         print(f"{mda_file=}")
         full_path_str=self.folderName()+'/'+mda_file
         self.setStatus(f"Selected file {full_path_str!r}.")      
         self._mdaFileName=mda_file
     
-    def setFolderPath(self, folder_path = DATA_FOLDER):
+    def setFolderPath(self, folder_name = DATA_FOLDER):
         """A folder was selected (from the open dialog)."""
-        # TODO: check for validity (folder exists? does it has mda in there?)
+        # TODO: check for validity (does it has mda in there?)
         # if len(catalog_name) == 0 or catalog_name not in self.server():
         #     if len(catalog_name) > 0:
         #         self.setStatus(f"Catalog {catalog_name!r} is not supported now.")
         #     return
-        folder_name = str(folder_path)
+
+        folder_path = Path(folder_name)
+
         self._folderPath = folder_path
         self._folderName = folder_name
         
-        if isinstance(folder_path,Path): 
+        # FIXME the is bellow is not longer very relevant since we now convert the str to path just above; need to check for validity above
+        if isinstance(folder_path,Path):   
             from .mda_folder import MDA_MVC
 
             # TODO: check if the folder has mda files in it 
             spec_name = MDA_SPEC_NAME
             self.spec_name.setText(spec_name)
+            # TODO do we really want to display that or is there a more relevant message to display? (eg valid folder?)
```

Comment:
Good question.  My intention was to display what was identified.  This message might be more important to the developer and less relevant to the end user.

Reply:
BUT, if the folder is not valid, there will be other indications, such as not content.

Reply:
What change are you requesting?

Reply:
Just responding to the question in the code.  Since is not the gemviz app, it's up to you how to handle.  Might not even be useful to report anything here.

Reply:
Yes, that was the point of my question/comment. Outside of the tiled server context, this field becomes irrelevant.